examples/wakaama/lwm2mclient.c

```diff
@@ -0,0 +1,646 @@
+ *    Christian Manal - Ported to RIOT OS
+ *
+ *******************************************************************************/
+
+/*
+ Copyright (c) 2013, 2014 Intel Corporation
+
+ Redistribution and use in source and binary forms, with or without
+ modification,
+ are permitted provided that the following conditions are met:
+
+     * Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+     * Redistributions in binary form must reproduce the above copyright notice,
+       this list of conditions and the following disclaimer in the documentation
+       and/or other materials provided with the distribution.
+     * Neither the name of Intel Corporation nor the names of its contributors
+       may be used to endorse or promote products derived from this software
+       without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED.
+ IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ OR
+ OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ OF
+ THE POSSIBILITY OF SUCH DAMAGE.
+
+ David Navarro <david.navarro@intel.com>
+ Bosch Software Innovations GmbH - Please refer to git log
+
+*/
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       Main loop for Wakaama example client.
+ *
+ * @author      Christian Manal <manal@uni-bremen.de>
+ *
+ * @}
+ */
+
+#include "dtlsconnection.h"
+#include "liblwm2m.h"
+#include "lwm2mclient.h"
+#include "lwm2mconfig.h"
+
+#include "periph/pm.h"
+
+#include <ctype.h>
+#include <errno.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#define MAX_PACKET_SIZE 1024
+#define DEFAULT_SERVER_IPV6 "[::1]"
+#define DEFAULT_SERVER_IPV4 "127.0.0.1"
+
+// Taken from examples/dtls-echo/dtls-server.c
+#define READER_QUEUE_SIZE (8U)
+
+static int g_quit = 0;
+
+#define OBJ_COUNT 4
+lwm2m_object_t *objArray[OBJ_COUNT];
+
+// only backup security and server objects
+#define BACKUP_OBJECT_COUNT 2
+lwm2m_object_t *backupObjectArray[BACKUP_OBJECT_COUNT];
+
+typedef struct __attribute__((packed)) {
+   lwm2m_object_t *securityObjP;
+   lwm2m_object_t *serverObject;
+   sock_udp_t sock;
+   dtls_connection_t *connList;
+   lwm2m_context_t *lwm2mH;
+   int addressFamily;
+} client_data_t;
+
+void handle_sigint(int signum) {
```

Comment:
In my opinion, this should be only for native and even then I would prefer to not have it there, but instead use a command (by example your `exit_cmd ()` command) which supports either [events](https://github.com/RIOT-OS/RIOT/blob/master/tests/events/main.c) or [IPC](http://riot-os.org/api/group__core__msg.html) .

At minimum, for native you should make this function equal to your `exit_cmd()`